job/job.go

```diff
@@ -86,3 +89,35 @@ func GetGenericJob(obj Object, config Config) (*batchv1.Job, error) {
 
 	return job, err
 }
+
+func getResourceRequirementsTemplate(r *corev1.ResourceRequirements) corev1.ResourceRequirements {
+	if r == nil {
+		return corev1.ResourceRequirements{
+			Requests: getDefaultResourceRequests(),
+			Limits:   getDefaultResourceLimits(),
+		}
+	}
+	if r.Limits == nil {
+		r.Limits = getDefaultResourceLimits()
+	}
+	if r.Requests == nil {
+		r.Requests = getDefaultResourceRequests()
+	}
+	return *r
+}
+
+func getDefaultResourceLimits() map[corev1.ResourceName]resource.Quantity {
+	li := make(map[corev1.ResourceName]resource.Quantity)
+	li[corev1.ResourceCPU] = resource.MustParse(cfg.Config.GlobalCPUResourceLimit)
+	li[corev1.ResourceMemory] = resource.MustParse(cfg.Config.GlobalMemoryResourceLimit)
+
+	return li
+}
+
+func getDefaultResourceRequests() map[corev1.ResourceName]resource.Quantity {
+	re := make(map[corev1.ResourceName]resource.Quantity)
+	re[corev1.ResourceCPU] = resource.MustParse(cfg.Config.GlobalCPUResourceRequest)
+	re[corev1.ResourceMemory] = resource.MustParse(cfg.Config.GlobalMemoryResourceRequest)
```

Comment:
Shouldn't there be some error handling? Users could decide to not specify CPU limits (leave empty), [as is recommended by upstream Kubernetes for production workload](https://learnk8s.io/production-best-practices). `MustParse` suggests there will be a panic on errors

Reply:
as far as I understood, we'll always have default: https://github.com/vshn/k8up/issues/143

Reply:
Hm, I wouldn't say so, but that's not entirely clear either :)
If we start defining default limits, this could break many existing backups out there. I would argue that "make it possible to define limits" doesn't mean we have to define defaults.
So I really think we should leave defaults empty in K8up v1alpha1, but we can revise those in K8up v1alpha2 (= v2 of K8up).
I'll clear up the story

Reply:
Done, is that more clear?